Exception/InternalError.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Hoa\Compiler\Exception;
+
+use LogicException;
+
+/**
+ * It probably points to some internal issue of the Hoa Compiler library.
+ * Regardless source of the bug, try to report about this exception to the library maintainers.
+ * Even if bug is yours, this exception must not happen.
+ */
+final class InternalError extends LogicException
+{
+    public function __construct($message, Exception $previous = null)
```

Comment:
Once the class extends a parent exception in the same directory, the constructor can be removed.